jetty-core/jetty-siwe/src/test/java/org/eclipse/jetty/security/siwe/SightInWithEthereumTokenTest.java

```diff
@@ -0,0 +1,237 @@
+// ========================================================================
+//
+
+package org.eclipse.jetty.security.siwe;
+
+import java.time.LocalDateTime;
+import java.util.function.Predicate;
+
+import org.eclipse.jetty.security.siwe.internal.EthereumUtil;
+import org.eclipse.jetty.security.siwe.internal.SignInWithEthereumParser;
+import org.eclipse.jetty.security.siwe.internal.SignInWithEthereumToken;
+import org.eclipse.jetty.security.siwe.util.EthereumCredentials;
+import org.eclipse.jetty.security.siwe.util.SignInWithEthereumGenerator;
+import org.eclipse.jetty.util.IncludeExcludeSet;
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class SightInWithEthereumTokenTest
+{
+    @Test
+    public void testInvalidVersion()
+    {
+        EthereumCredentials credentials = new EthereumCredentials();
+        LocalDateTime issuedAt = LocalDateTime.now();
+        String message = SignInWithEthereumGenerator.generateMessage(
+            null,
+            "example.com",
+            credentials.getAddress(),
+            "hello this is the statement",
+            "https://example.com",
+            "2",
+            "1",
+            EthereumUtil.createNonce(),
+            issuedAt,
+            null, null, null, null
+        );
+
+        SignedMessage signedMessage = credentials.signMessage(message);
+        SignInWithEthereumToken siwe = SignInWithEthereumParser.parse(message);
+        assertNotNull(siwe);
+
+        Throwable error = assertThrows(Throwable.class, () ->
+            siwe.validate(signedMessage, null, null, null, null));
+        assertThat(error.getMessage(), containsString("unsupported version"));
+    }
+
+    @Test
+    public void testExpirationTime()
+    {
+        EthereumCredentials credentials = new EthereumCredentials();
+        LocalDateTime issuedAt = LocalDateTime.now().minusSeconds(10);
+        LocalDateTime expiry = LocalDateTime.now();
+        String message = SignInWithEthereumGenerator.generateMessage(
+            null,
+            "example.com",
+            credentials.getAddress(),
+            "hello this is the statement",
+            "https://example.com",
+            "1",
+            "1",
+            EthereumUtil.createNonce(),
+            issuedAt,
+            expiry,
+            null, null, null
+        );
+
+        SignedMessage signedMessage = credentials.signMessage(message);
+        SignInWithEthereumToken siwe = SignInWithEthereumParser.parse(message);
+        assertNotNull(siwe);
+
+        Throwable error = assertThrows(Throwable.class, () ->
+            siwe.validate(signedMessage, null, null, null, null));
+        assertThat(error.getMessage(), containsString("expired SIWE message"));
+    }
+
+    @Test
+    public void testNotBefore()
```

Comment:
Umm, not before what? Can you make the name of the test more descriptive.

Reply:
There is a field in the SIWE Message called `not-before`.

I could rename the test to something like `testNotYetValid` but it brings it out of line with the naming of all the other tests which are named directly with the field of the SIWE Message.